tidy3d/components/source.py

```diff
@@ -465,6 +465,29 @@ def _dir_vector(self) -> Tuple[float, float, float]:
         return None
 
 
+class BroadbandSource(Source, ABC):
+    """A source with frequency dependent field distributions."""
+
+    num_freqs: pydantic.NonNegativeInt = pydantic.Field(
+        1,
+        title="Number of Frequency Points",
+        description="Number of points to approximate the frequency dependence of injected field.",
+    )
+
+    @cached_property
+    def frequency_grid(self) -> np.ndarray:
+        """A Chebyshev grid used to approximate frequency dependence."""
+        fmin, fmax = self.source_time.frequency_range(4)
+        return 0.5 * (fmin + fmax) + 0.5 * (fmax - fmin) * np.cos(
+            0.5 * np.pi * (2 * np.flip(np.arange(self.num_freqs)) + 1) / self.num_freqs
+        )
+
+    def map_frequencies(self, freq_grid) -> np.ndarray:
+        """Map frequency values into (-1,1) interval."""
+        fmin, fmax = self.source_time.frequency_range(4)
```

Comment:
can we add the keyword argument explicitly? Ie

```py
fmin, fmax = self.source_time.frequency_range(num_fwidth=4)
```
I had to look up what `frequency_range` accepted because I forgot, doing it this way will make it a bit more readable.  (note that 4 is the default value, so alternatively we could just remove the argument here)